subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveBeforeExecutionStateTaskExecuter.java

```diff
@@ -113,24 +113,24 @@ private BeforeExecutionState createExecutionState(TaskInternal task, TaskPropert
         );
     }
 
-    private ImmutableSortedMap<String, CurrentFileCollectionFingerprint> createFilteredOutputFingerprints(@Nullable AfterPreviousExecutionState afterPreviousExecutionState, ImmutableSortedMap<String, FileSystemSnapshot> outputFileSnapshots, boolean hasOverlappingOutputs) {
-        ImmutableSortedMap<String, FileCollectionFingerprint> previousOutputs = afterPreviousExecutionState == null ? ImmutableSortedMap.of() : afterPreviousExecutionState.getOutputFileProperties();
+    private ImmutableSortedMap<String, CurrentFileCollectionFingerprint> createFilteredOutputFingerprints(@Nullable AfterPreviousExecutionState afterPreviousExecutionState, ImmutableSortedMap<String, FileSystemSnapshot> beforeExecutionOutputSnapshots, boolean hasOverlappingOutputs) {
+        ImmutableSortedMap<String, FileCollectionFingerprint> afterLastExecutionFingerprints = afterPreviousExecutionState == null ? ImmutableSortedMap.of() : afterPreviousExecutionState.getOutputFileProperties();
 
         return ImmutableSortedMap.copyOfSorted(
-            Maps.transformEntries(outputFileSnapshots, (key, outputSnapshot) -> {
-                    FileCollectionFingerprint previousOutputFingerprint = previousOutputs.get(key);
-                    return previousOutputFingerprint == null
+            Maps.transformEntries(beforeExecutionOutputSnapshots, (key, beforeExecutionOutputSnapshot) -> {
+                    FileCollectionFingerprint afterLastExecutionFingerprint = afterLastExecutionFingerprints.get(key);
+                    return afterLastExecutionFingerprint == null
```

Comment:
We have been using `afterPreviousExecution` everywhere else, I think we should do it here, too. That said, I'm already basing some changes on top of this that will correct it, so please don't change.